src/utils/time.ts

```diff
@@ -5,4 +5,8 @@
  */
 export function sleep(durationInMillisecond): Promise<void> {
   return new Promise(resolve => setTimeout(resolve, durationInMillisecond));
+}
+
+export function getCurrentDateInHighPrecision(): string {
+  return (new Date().toISOString().replace('Z',Math.floor(Math.random()*(999-100+1)+100).toString())).toString();
```

Comment:
Instead of randomly generating microseconds, I think the following would suffice for this PR:
```suggestion
  return new Date().toISOString().replace('Z','000');
```

If you want to go the extra miles, you can either:
1. look into using npm package that are BOTH browser and nodejs compatible; or
2. look into `performance.timeOrigin` and `performance.now()`

But that's optional as you've done the painful work of fixing all references already!